app/main.py

```diff
@@ -1,8 +1,35 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name, age):
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    def set_spouse(self, spouse_name):
+        if spouse_name:
+            spouse = Person.people.get(spouse_name)
+            if spouse:
+                if hasattr(self, "wife") or hasattr(self, "husband"):
+                    return  # Уже установлен супруг
+                if self.name not in ["Rachel", "Monica"]:
```

Comment:
The condition `if self.name not in ["Rachel", "Monica"]:` is hardcoded and assumes that only these two names should have a husband. This is not scalable and could lead to errors if more names need to be added or if the logic changes. Consider using a more flexible approach, such as a configuration or a property of the `Person` class, to determine the spouse type.